lmp/tokenizer/_base_tokenizer.py

```diff
@@ -486,9 +486,22 @@ def batch_decode(
                 tokens except unknown word's token.
                 See class docstring for more details on special tokens.
 
+        Raises:
+            TypeError:
+                When `batch_token_ids` is not instance of `List[List[int]]`.
+                When `remove_special_tokens` is not instance of `bool`.
+
         Returns:
             Batch of sequence decoded from `batch_token_ids`.
         """
+        # Type check.
+        if not isinstance(batch_token_ids, List):
+            raise TypeError(
+                '`batch_token_ids` must be instance of `List[List[int]]`.')
```

Comment:
new line before right parenthesis

Reply:
First do `batch_token_ids = list(batch_token_ids)`, then do `batch_token_ids = [list(token_ids) for token_ids in batch_token_ids]`, finally use `for` loop to check each element in each element of `batch_token_ids` is an instance of `int`